src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -5701,34 +5701,72 @@ internal static List<CompletionResult> CompleteMember(CompletionContext context,
                     targetExpr = nextToLastAst as ExpressionAst;
                 }
             }
-            else if (lastAst.Parent is MemberExpressionAst)
+            else if (lastAst.Parent is MemberExpressionAst parentAsMemberExpression)
             {
-                // If 'targetExpr' has already been set, we should skip this step. This is for some member completion
-                // cases in ISE. In ISE, we may add a new statement in the middle of existing statements as follows:
-                //     $xml = New-Object Xml
-                //     $xml.
-                //     $xml.Save("C:\data.xml")
-                // In this example, we add $xml. between two existing statements, and the 'lastAst' in this case is
-                // a MemberExpressionAst '$xml.$xml', whose parent is still a MemberExpressionAst '$xml.$xml.Save'.
-                // But here we DO NOT want to re-assign 'targetExpr' to be '$xml.$xml'. 'targetExpr' in this case
-                // should be '$xml'.
```

Comment:
Can you please keep this comment? We don't mind ISE anymore, but the same scenario can happen in VSCode. I see the code that covers this scenario is still there.

Reply:
I've added it back. I think I misunderstood the the comment the first time and thought it was explaining a scenario where the if statement would be true, when in reality it's explaining why it has that check.